src/main/java/org/broadinstitute/dsm/careevolve/Covid19OrderRegistrar.java

```diff
@@ -60,184 +66,78 @@ public Covid19OrderRegistrar(String endpoint,
+                try {
+                    Thread.sleep(retryWaitMillis);
+                } catch (InterruptedException interruptedException) {
+                    logger.error("Interrupted while waiting for CE order retry for patient " + patientId, e);
+                }
+            }
+            numAttempts++;
+        } while (numAttempts < maxRetries && !orderSucceeded);
+
+        if (!orderSucceeded) {
+            String exceptionsText = StringUtils.join(orderExceptions, "\n");
+            throw new CareEvolveException("Could not order test for " + patientId + " after " + maxRetries + ":\n" + exceptionsText);
+        }
+
+        if (StringUtils.isNotBlank(orderResponse.getError())) {
+            throw new CareEvolveException("Order for participant " + patientId + " with handle  " + orderResponse.getHandle() + " placed with error " + orderResponse.getError());
+        }
+        return orderResponse;
+    }
+
     /**
      * Places an order in CareEvolve
-     * @param auth the API credentials
+     *
+     * @param auth             the API credentials
      * @param participantHruid hruid for the participant
-     * @param kitLabel The label on the swab.  Corresponds to ORC-2 and GP sample_id
-     * @param kitId an identifier that will show up in Birch to help
-     *              associate the result back to the proper kit
-     * @param kitPickupTime the time at which the kit was picked
-     *                      up from the participant
+     * @param kitLabel         The label on the swab.  Corresponds to ORC-2 and GP sample_id
+     * @param kitId            an identifier that will show up in Birch to help
+     *                         associate the result back to the proper kit
+     * @param kitPickupTime    the time at which the kit was picked
+     *                         up from the participant
      */
-    public  OrderResponse orderTest(Authentication auth, String participantHruid, String kitLabel,
-                                          String kitId, Instant kitPickupTime) throws CareEvolveException {
+    public OrderResponse orderTest(Authentication auth, String participantHruid, String kitLabel,
+                                   String kitId, Instant kitPickupTime) throws CareEvolveException {
         if (kitPickupTime == null) {
             throw new CareEvolveException("Cannot place order for " + kitLabel + " without a pickup time");
         }
 
-        DDPInstance instance = DDPInstance.getDDPInstanceWithRole("testboston", DBConstants.HAS_KIT_REQUEST_ENDPOINTS);
-
-        Map<String, String> queryConditions = new HashMap<>();
-        queryConditions.put("ES", " AND profile.hruid = '" + participantHruid +"'");
-        List<ParticipantWrapper> participants = ParticipantWrapper.getFilteredList(instance, queryConditions);
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole("testboston", DBConstants.HAS_KIT_REQUEST_ENDPOINTS);
```

Comment:
is there a reason why we check if the instance has that role?

Reply:
@pegahtah if the role is not used, can you please change it to just getDDPInstance